### PR TITLE
Remove legacy FIPS_mode functions

### PR DESCRIPTION
**Pull request by slontis**
This is the alternative to #11075 which did a deprecation.

##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by paulidale** [comment]
OMC vote is in progress to decide between this and #11075 so putting a hold on both.

**Comment by paulidale** [comment]
Block removed.  These functions are to be removed.

An independent review is required.

File: CHANGES.md [hunk 1]
```diff
@@ -18,12 +18,19 @@ OpenSSL Releases
  - [OpenSSL 1.0.1](#openssl-101)
  - [OpenSSL 1.0.0](#openssl-100)
  - [OpenSSL 0.9.x](#openssl-09x)
-
+)
```
Comment (mattcaswell) [inline CHANGES.md:21]: Oh...wait...what is this close paren doing here?

Comment (slontis) [reply CHANGES.md]: I think I was distracted by the heart attack that this file seems to give my editor. Fixed now.

Comment (mspncp) [reply CHANGES.md]: Oh c'mon, you're kiddin' me! The (minimalistic) markdown was designed to not only look awesome in the browser but also to remain readable in raw ASCII format, see #10545.  At least I tried very hard to keep my promise.

Comment (richsalz) [reply CHANGES.md]: Perhaps running markdownlint on all ```.md``` files would be a fun project for someone to add to the CI's?

**Comment by slontis** [comment]
Thanks for everyone's input. Merged to master.

**Pull request closed without merging**